storage/gc_queue.go

```diff
@@ -236,27 +236,20 @@ func (gcq *gcQueue) process(now proto.Timestamp, repl *Replica) error {
 	wg.Wait()
 
 	// Resolve all intents.
-	// TODO(spencer): use a batch here when available.
+	var intents []proto.Intent
 	for id, txn := range txnMap {
 		if txn.Status != proto.PENDING {
-			// The transaction was successfully pushed, so resolve the intents.
-			for _, key := range intentMap[id] {
-				resolveArgs := &proto.ResolveIntentRequest{
-					RequestHeader: proto.RequestHeader{
-						Timestamp: now,
-						Key:       key,
-						User:      security.RootUser,
-						Txn:       txn,
-					},
-				}
-				if _, err := repl.AddCmd(repl.context(), resolveArgs); err != nil {
-					log.Warningf("resolve of key %q failed: %s", key, err)
-					updateOldestIntent(txn.OrigTimestamp.WallTime)
-				}
+			for _, intent := range intentMap[id] {
+				intent.Txn = *txn
+				intents = append(intents, intent)
 			}
 		}
 	}
 
+	if len(intents) > 0 {
+		repl.resolveIntents(repl.context(), intents)
```

Comment:
yes, I'm planning to have it send out another GC request (since it's range-local space, so you can't add it to the other one). I have a WIP but got sidetracked doing other stuff today. Will put something up soon though.